Pose estimation v2

KangarooKoala opened the pull request:
Switched from 2910 to WPILib geometry classes
Added method to convert from `Vector2` to `Translation2d`
Reformatted doc comments
Specified field-centric frame of reference
Switched from hub-centric to field-centric
Added null checks for gyro angle
Added `getGyroscopeYaw()` and `getFieldCentricTurretAngle()`
Removed `getVisionGyroRobotPoseRelativeToStart()`
Specified method frames of reference

KangarooKoala committed:
Switch from 2910 to WPILib geometry classes
Changed from inches to meters
Extracted turret relative velocity to separate method
Reformatted pose doc comments
Added method to convert from Vector2 to Translation2d to GeoConvertor

KangarooKoala committed:
Pose frames of reference
Added doc comment to TargetLocalizer specifying field-centric poses
Switched back to inches (should discuss and finalize)
Switched from hub-centric to field-centric
Added null checks for gyro angle
Added getGyroscopeYaw() and getFieldCentricTurretAngle()
Removed getVisionGyroRobotPoseRelativeToStart()
Specified frames of reference
Formatted some doc comments

KangarooKoala commented:
Next update is to combine drivebase pose and vision+gyro pose with [`SwerveDrivePoseEstimator`](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/math/estimator/SwerveDrivePoseEstimator.html), but since that would involve somewhat intrusive modifications to `DrivebaseSubsystem` (possibly even moving all of the pose stuff to `TargetLocalizer`), I'm holding off on it until these changes get merged in.

KangarooKoala commented:
Also, we should decide if we're going to use inches or meters.

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Added correction for different start poses
Made startingPose and gyroAdjustmentAngle non-final
Added resetPose() and resetPoseFromAutoStartPose()
Reset target localizer pose in autonomousInit()

TAKBS2412 commented on src/main/java/frc/team2412/robot/subsystem/DrivebaseSubsystem.java:
```diff
@@ -232,9 +233,14 @@ public Vector2 getGyroscopeXY() {
+    public Rotation2d getGyroscopeUnadjustedAngle() {
         synchronized (sensorLock) {
-            return gyroscope.getUnadjustedAngle();
+            return Rotation2d.fromDegrees(gyroscope.getUnadjustedAngle().toDegrees());
```
Could we use the constructor with radians instead of having degrees appear twice?

KangarooKoala replied:
Yes, but I thought that using `fromDegrees()` and `toDegrees()` would be clearer.

TAKBS2412 reviewed (commented):
Haven't examined the math in detail yet will look later.

KangarooKoala committed:
Moved constants to appropriate subsystems
Also fixed typo in docs

KangarooKoala commented on src/main/java/frc/team2412/robot/subsystem/TargetLocalizer.java:
```diff
@@ -99,30 +112,48 @@ public double getVisionYaw() {
+    /**
+     * Returns the gyroscope's yaw.
+     *
+     * @return The gyroscope's yaw without the adjustment angle (positive rotation is counterclockwise).
```
Yes, the reason `getVisionYaw()`, `getTargetYaw()`, and `getTurretAngle()` have positive rotation as clockwise is to avoid interfering with other code (Especially feedforward math, though shooter and shooter vision would also be affected).

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation
Used turretAngularFF instead of TURRET_ANGULAR_FF

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation
Fix typo in getDepthVelocity()
Added check if the limelight has a target before filtering

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Add VisionUtil
Separate pose estimation logic
Change getTurretAngle() to be CCW-positive

KangarooKoala committed:
Misc. fixes and clean-ups
Remove doc comment parameter for removed parameter
Change ROBOT_CENTRIC_TURRET_CENTER to +X axis forward (intake side)
Remove unused startingPose
Ignore turret bias in pose estimation

KangarooKoala committed:
Merge branch 'main' of https://github.com/robototes/RapidReact2022 into pose-estimation

KangarooKoala committed:
Improve doc comments and correct naming
